pkg/plugin/run.go

```diff
@@ -175,17 +177,33 @@ func (e *AutoscaleEnforcer) handleAgentRequest(
 	resp := api.PluginResponse{
 		Permit:      permit,
 		Migrate:     migrateDecision,
-		ComputeUnit: *node.computeUnit,
+		ComputeUnit: getComputeUnitForResponse(node, supportsFractional),
 	}
 	pod.mostRecentComputeUnit = node.computeUnit // refer to common allocation
 	return &resp, 200, nil
 }
 
+// getComputeUnitForResponse tries to return compute unit that the agent supports
+// if we have a fractional CPU but the agent does not support it we multiply the result
+func getComputeUnitForResponse(node *nodeState, supportsFractional bool) api.Resources {
+	computeUnit := *node.computeUnit
+	if !supportsFractional {
+		initialCU := computeUnit
+		i := uint16(2)
+		for ; computeUnit.VCPU < 1000; i++ {
```

Comment:
```suggestion
		for i := uint16(2); computeUnit.VCPU % 1000 != 0; i++ {
```

As currently written, this will be incorrect if `1000 % computeUnit.VCPU != 0` (e.g. if `VCPU` is 750m, we will see that `1500 >= 1000` and exit, which would still give the autoscaler-agent a non-integer value)

Another option, food for thought: Multiply `computeUnit` by the least common multiple of `computeUnit.VCPU` and 1000. Maybe overkill (LCM is annoying); "dumb" solution is ok.